### PR TITLE
Only export the generic traced error version

### PR DESCRIPTION
**Pull request by yaahc**

### CODE REVIEW

File: tracing-error/src/lib.rs [hunk 1]
```diff
@@ -16,23 +16,26 @@
...
 //!
-//! - `stack-error` - Enables an experimental version of [`TracedError`] that
-//! statically wraps an error type without using heap allocations. The contained
-//! [`SpanTrace`] can still be extracted from behind a
-//! [`std::error::Error`] trait object via type erasure.
+//! - `traced-error` - Enables a generic error type that bundles a `SpanTrace`
+//! with an inner error type. This is a good match for the error-kind pattern
+//! where you have an error type with an inner enum of error variants and you
+//! would like to capture a span trace that can be extracted during printing
+//! without formatting the span trace as part of your display impl.
```
Comment (hawkw) [inline tracing-error/src/lib.rs]: I might just put the usage details here in the API docs for the type?

File: tracing-error/src/lib.rs [hunk 2]
```diff
@@ -16,23 +16,26 @@
...
 //!
 //! *Compiler support: requires `rustc` 1.39+*
 //!
 //! ## Feature Flags
 //!
-//! - `stack-error` - Enables an experimental version of [`TracedError`] that
-//! statically wraps an error type without using heap allocations. The contained
-//! [`SpanTrace`] can still be extracted from behind a
-//! [`std::error::Error`] trait object via type erasure.
+//! - `traced-error` - Enables a generic error type that bundles a `SpanTrace`
```
Comment (hawkw) [inline tracing-error/src/lib.rs]: I think this should link to the `TracedError` struct.

File: tracing-error/src/lib.rs [hunk 3]
```diff
@@ -16,23 +16,26 @@
...
+//! without formatting the span trace as part of your display impl.
+//!
+//! Enabling the above feature also brings in the following traits
+//!
+//! * [`InstrumentResult`] and [`InstrumentError`] extension traits, which
+//! provide an [`in_current_span()`] method for bundling errors with a
+//! [`SpanTrace`].
+//!
+//! * [`ExtractSpanTrace`] extension trait, for extracting `SpanTrace`s from
+//! behind `dyn Error` trait objects.
```
Comment (hawkw) [inline tracing-error/src/lib.rs]: I think this should all just be indented under the `traced-error` bullet point.

File: tracing-error/src/lib.rs [hunk 4]
```diff
@@ -196,98 +200,23 @@
...
+#[cfg(feature = "traced-error")]
+mod error;
 mod layer;
-#[cfg(feature = "stack-error")]
-mod stack_error;
 
 pub use self::backtrace::{SpanTrace, SpanTraceStatus};
-#[cfg(not(feature = "stack-error"))]
-pub use self::heap_error::TracedError;
+#[cfg(feature = "traced-error")]
```
Comment (hawkw) [inline tracing-error/src/lib.rs]: I think this needs a `doc-cfg` attribute.

File: tracing-error/src/lib.rs [hunk 5]
```diff
@@ -172,6 +175,7 @@
 //! [subscriber layer]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html
 //! [`tracing`]: https://docs.rs/tracing
 //! [`std::error::Error`]: https://doc.rust-lang.org/stable/std/error/trait.Error.html
+#![cfg_attr(docsrs, feature(doc_cfg))]
```
Comment (hawkw) [inline tracing-error/src/lib.rs]: remember that this needs to be added to cargo.toml as well:
https://github.com/tokio-rs/tracing/blob/86d6c1b954a95d4185e8ab8e427836a1f1820934/tracing/Cargo.toml#L76-L78

File: tracing-error/src/lib.rs (hunk 4 above)
Comment (yaahc) [reply tracing-error/src/lib.rs]: I tried putting it here first and it had no effect so I moved it to the definitions of the structs / traits in error.rs to get it to actually show up in the docs

File: tracing-error/src/lib.rs (hunk 5 above)
Comment (yaahc) [reply tracing-error/src/lib.rs]: Already there at the bottom

**Pull request merged**